### PR TITLE
HOSTEDCP-1614: Use marketplace images in aro-hcp compatible jobs

### PR DESCRIPTION
**Pull request by fxierh**
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/55312/rehearse-55312-periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7/1821506793973485568

### CODE REVIEW

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.17-periodics-e2e-azure-ovn-conformance

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

File: ci-operator/step-registry/cucushift/hypershift-extended/nodepool-image/azure/get-marketplace-image/cucushift-hypershift-extended-nodepool-image-azure-get-marketplace-image-commands.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+echo "Extracting OCP version from release image"
+OCP_MAJOR_VERSION=$(oc adm release info "$RELEASE_IMAGE_LATEST" -a "$WRITABLE_PULL_SECRET" -o json | jq -r '.metadata.version' | cut -d . -f 1)
+OCP_MINOR_VERSION=$(oc adm release info "$RELEASE_IMAGE_LATEST" -a "$WRITABLE_PULL_SECRET" -o json | jq -r '.metadata.version' | cut -d . -f 2)
+
+echo "Extracting Azure marketplace image info"
+az vm image list \
+    --architecture "$HYPERSHIFT_AZURE_MARKETPLACE_IMAGE_ARCH" \
+    --publisher "$HYPERSHIFT_AZURE_MARKETPLACE_IMAGE_PUBLISHER" \
+    --offer "$HYPERSHIFT_AZURE_MARKETPLACE_IMAGE_OFFER" \
+    --sku "aro_${OCP_MAJOR_VERSION}${OCP_MINOR_VERSION}" \
```
Comment (bryan-cox) [inline ci-operator/step-registry/cucushift/hypershift-extended/nodepool-image/azure/get-marketplace-image/cucushift-hypershift-extended-nodepool-image-azure-get-marketplace-image-commands.sh:40]: I don't know that this is standardized. I just look and saw several variations `415-v2, 417-arm, aro_415` etc.

Comment (fxierh) [reply ci-operator/step-registry/cucushift/hypershift-extended/nodepool-image/azure/get-marketplace-image/cucushift-hypershift-extended-nodepool-image-azure-get-marketplace-image-commands.sh:40]: I'd like to dynamically select the SKU based on the HC's version and use the `aro_XY` naming for now.

If the naming convention changes in the future, we can introduce an environment variable to control the naming scheme, allowing for options like `aro_XY`, `XY-arm`, etc.

Comment (bryan-cox) [reply ci-operator/step-registry/cucushift/hypershift-extended/nodepool-image/azure/get-marketplace-image/cucushift-hypershift-extended-nodepool-image-azure-get-marketplace-image-commands.sh:40]: Ok that's fine. Just wanted to make sure you were aware there were other variations.

Comment (fxierh) [reply ci-operator/step-registry/cucushift/hypershift-extended/nodepool-image/azure/get-marketplace-image/cucushift-hypershift-extended-nodepool-image-azure-get-marketplace-image-commands.sh:40]: Yup, thanks for the info

**Comment by fxierh** [comment]
/cc @liangxia 

Could you PTAL ?

**Pull request merged**